packages/awsfirehose/data_stream/metrics/routing_rules.yml

```
@@ -1,112 +1,112 @@
 - source_dataset: aws.cloudwatch
   rules:
     - target_dataset: aws.apigateway_metrics
-      if: ctx['aws.cloudwatch.namespace'] != null &&  ctx['aws.cloudwatch.namespace'] == "AWS/ApiGateway"
+      if: ctx?.aws?.cloudwatch?.namespace == "AWS/ApiGateway"
```

```suggestion
      if: ctx.aws?.cloudwatch?.namespace == "AWS/ApiGateway"
```
Similar below; `ctx` is always present.

---

oh yes...! Good catch thank you!!